net/sqlwriter.c

```diff
@@ -177,19 +204,26 @@ static void sql_flush_cb(int fd, short what, void *arg)
     while (evbuffer_get_length(writer->wr_buf)) {
         if ((n = wr_evbuffer(writer, fd)) <= 0) break;
         writer->sent_at = time(NULL);
+        update_writer_state(writer, WRITE_SUCCEEDED);
     }
     if (evbuffer_get_length(writer->wr_buf) == 0) {
         sql_disable_flush(writer);
-    } else if (n <= 0 && errno != EAGAIN && errno != EWOULDBLOCK) {
-        writer->bad = 1;
-        event_del(writer->flush_ev);
+    } else if (n <= 0) {
+        if (errno == EAGAIN || errno == EWOULDBLOCK) {
+            update_writer_state(writer, WRITE_BLOCKED);
+        } else {
+            writer->bad = 1;
+            event_del(writer->flush_ev);
+            update_writer_state(writer, WRITE_FAILED);
+        }
```

Comment:
Nitpick: Maybe the other state changes to `writer` that happen here could also go into `update_writer_state`? (like setting `writer->bad` and `writer->sent_at`)